{
 "PRed": [
  {
   "PR": {
    "number": 1,
    "state": "closed"
   },
   "data": {
    "bot_rerun": false,
    "migrator_name": "CompilerRebuild",
    "migrator_version": 1,
    "name": "Python 3.7, GCC 7, R 3.5.1, openBLAS 0.3.2"
   },
   "keys": [
    "bot_rerun",
    "migrator_name",
    "migrator_version",
    "name"
   ]
  },
  {
   "PR": {
    "number": 3,
    "state": "closed"
   },
   "data": {
    "bot_rerun": false,
    "migrator_name": "MigrationYaml",
    "migrator_object_version": 1,
    "migrator_version": 0,
    "name": "python38"
   },
   "keys": [
    "bot_rerun",
    "migrator_name",
    "migrator_object_version",
    "migrator_version",
    "name"
   ]
  },
  {
   "PR": {
    "number": 4,
    "state": "closed"
   },
   "data": {
    "bot_rerun": false,
    "migrator_name": "MigrationYaml",
    "migrator_object_version": 1,
    "migrator_version": 0,
    "name": "pypy"
   },
   "keys": [
    "bot_rerun",
    "migrator_name",
    "migrator_object_version",
    "migrator_version",
    "name"
   ]
  },
  {
   "PR": {
    "number": 5,
    "state": "closed"
   },
   "data": {
    "bot_rerun": false,
    "migrator_name": "MigrationYaml",
    "migrator_object_version": 2,
    "migrator_version": 0,
    "name": "python39"
   },
   "keys": [
    "bot_rerun",
    "migrator_name",
    "migrator_object_version",
    "migrator_version",
    "name"
   ]
  },
  {
   "PR": {
    "number": 6,
    "state": "closed"
   },
   "data": {
    "bot_rerun": false,
    "migrator_name": "MigrationYaml",
    "migrator_object_version": 1,
    "migrator_version": 0,
    "name": "pypy37"
   },
   "keys": [
    "bot_rerun",
    "migrator_name",
    "migrator_object_version",
    "migrator_version",
    "name"
   ]
  },
  {
   "PR": {
    "number": 7,
    "state": "closed"
   },
   "data": {
    "bot_rerun": false,
    "migrator_name": "MigrationYaml",
    "migrator_object_version": 1,
    "migrator_version": 0,
    "name": "pypy37-windows"
   },
   "keys": [
    "bot_rerun",
    "migrator_name",
    "migrator_object_version",
    "migrator_version",
    "name"
   ]
  },
  {
   "PR": {
    "number": 8,
    "state": "closed"
   },
   "data": {
    "bot_rerun": false,
    "migrator_name": "MigrationYaml",
    "migrator_object_version": 1,
    "migrator_version": 0,
    "name": "python310"
   },
   "keys": [
    "bot_rerun",
    "migrator_name",
    "migrator_object_version",
    "migrator_version",
    "name"
   ]
  },
  {
   "PR": {
    "number": 9,
    "state": "closed"
   },
   "data": {
    "bot_rerun": false,
    "migrator_name": "MigrationYaml",
    "migrator_object_version": 1,
    "migrator_version": 0,
    "name": "pypy38"
   },
   "keys": [
    "bot_rerun",
    "migrator_name",
    "migrator_object_version",
    "migrator_version",
    "name"
   ]
  },
  {
   "PR": {
    "number": 10,
    "state": "closed"
   },
   "data": {
    "bot_rerun": false,
    "migrator_name": "MigrationYaml",
    "migrator_object_version": 1,
    "migrator_version": 0,
    "name": "python311"
   },
   "keys": [
    "bot_rerun",
    "migrator_name",
    "migrator_object_version",
    "migrator_version",
    "name"
   ]
  },
  {
   "PR": {
    "labels": [
     {
      "name": "automerge"
     }
    ],
    "number": 11,
    "state": "closed"
   },
   "data": {
    "bot_rerun": false,
    "migrator_name": "MigrationYaml",
    "migrator_object_version": 1,
    "migrator_version": 0,
    "name": "python312"
   },
   "keys": [
    "bot_rerun",
    "migrator_name",
    "migrator_object_version",
    "migrator_version",
    "name"
   ]
  }
 ],
 "bad": {
  "exception": "Error running migrate-feedstock in container - error VersionMigrationError raised:\n'The recipe did not change in the version migration, a URL did not hash, or there is jinja2 syntax the bot cannot handle!\n\nPlease check the URLs in your recipe with version '29.0.0' to make sure they exist!\n\nWe also found the following errors:\n\n - We found a problem parsing the recipe for version '29.0.0': \n\nParserError('while parsing a block mapping',   in \"<unicode string>\", line 32, column 3:\n      home: https://github.com/cucumbe ... \n      ^ (line: 32), \"expected <block end>, but found '<scalar>'\",   in \"<unicode string>\", line 35, column 30:\n      license_file: '<{ environ['RECIPE_DIR'] }}/LICENSE.txt'\n                                 ^ (line: 35))\n\ntraceback:\n  File \"/opt/autotick-bot/conda_forge_tick/update_recipe/version.py\", line 415, in update_version\n    cmeta = CondaMetaYAML(raw_meta_yaml)\n            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/opt/autotick-bot/conda_forge_tick/recipe_parser/_parser.py\", line 525, in __init__\n    self.meta = self._parser.load(\"\".join(lines))\n                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/opt/conda/envs/cf-scripts/lib/python3.11/site-packages/ruamel/yaml/main.py\", line 451, in load\n    return constructor.get_single_data()\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/opt/conda/envs/cf-scripts/lib/python3.11/site-packages/ruamel/yaml/constructor.py\", line 114, in get_single_data\n    node = self.composer.get_single_node()\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/opt/conda/envs/cf-scripts/lib/python3.11/site-packages/ruamel/yaml/composer.py\", line 72, in get_single_node\n    document = self.compose_document()\n               ^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/opt/conda/envs/cf-scripts/lib/python3.11/site-packages/ruamel/yaml/composer.py\", line 94, in compose_document\n    node = self.compose_node(None, None)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/opt/conda/envs/cf-scripts/lib/python3.11/site-packages/ruamel/yaml/composer.py\", line 130, in compose_node\n    node = self.compose_mapping_node(anchor)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/opt/conda/envs/cf-scripts/lib/python3.11/site-packages/ruamel/yaml/composer.py\", line 211, in compose_mapping_node\n    item_value = self.compose_node(node, item_key)\n                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/opt/conda/envs/cf-scripts/lib/python3.11/site-packages/ruamel/yaml/composer.py\", line 130, in compose_node\n    node = self.compose_mapping_node(anchor)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/opt/conda/envs/cf-scripts/lib/python3.11/site-packages/ruamel/yaml/composer.py\", line 204, in compose_mapping_node\n    while not self.parser.check_event(MappingEndEvent):\n              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/opt/conda/envs/cf-scripts/lib/python3.11/site-packages/ruamel/yaml/parser.py\", line 141, in check_event\n    self.current_event = self.state()\n                         ^^^^^^^^^^^^\n  File \"/opt/conda/envs/cf-scripts/lib/python3.11/site-packages/ruamel/yaml/parser.py\", line 614, in parse_block_mapping_key\n    raise ParserError(\n\n'",
  "traceback": [
   "Traceback (most recent call last):",
   "  File \"/home/runner/work/cf-scripts/cf-scripts/cf-scripts/conda_forge_tick/auto_tick.py\", line 567, in _run_migrator_on_feedstock_branch",
   "    migrator_uid, pr_json = run(",
   "                            ^^^^",
   "  File \"/home/runner/work/cf-scripts/cf-scripts/cf-scripts/conda_forge_tick/auto_tick.py\", line 211, in run",
   "    migration_run_data = run_migration(",
   "                         ^^^^^^^^^^^^^^",
   "  File \"/home/runner/work/cf-scripts/cf-scripts/cf-scripts/conda_forge_tick/migration_runner.py\", line 63, in run_migration",
   "    return run_migration_containerized(",
   "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^",
   "  File \"/home/runner/work/cf-scripts/cf-scripts/cf-scripts/conda_forge_tick/migration_runner.py\", line 157, in run_migration_containerized",
   "    data = run_container_task(",
   "           ^^^^^^^^^^^^^^^^^^^",
   "  File \"/home/runner/work/cf-scripts/cf-scripts/cf-scripts/conda_forge_tick/utils.py\", line 286, in run_container_task",
   "    raise ContainerRuntimeError(",
   "conda_forge_tick.utils.ContainerRuntimeError: Error running migrate-feedstock in container - error VersionMigrationError raised:",
   "'The recipe did not change in the version migration, a URL did not hash, or there is jinja2 syntax the bot cannot handle!",
   "",
   "Please check the URLs in your recipe with version '29.0.0' to make sure they exist!",
   "",
   "We also found the following errors:",
   "",
   " - We found a problem parsing the recipe for version '29.0.0': ",
   "",
   "ParserError('while parsing a block mapping',   in \"<unicode string>\", line 32, column 3:",
   "      home: https://github.com/cucumbe ... ",
   "      ^ (line: 32), \"expected <block end>, but found '<scalar>'\",   in \"<unicode string>\", line 35, column 30:",
   "      license_file: '<{ environ['RECIPE_DIR'] }}/LICENSE.txt'",
   "                                 ^ (line: 35))",
   "",
   "traceback:",
   "  File \"/opt/autotick-bot/conda_forge_tick/update_recipe/version.py\", line 415, in update_version",
   "    cmeta = CondaMetaYAML(raw_meta_yaml)",
   "            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^",
   "  File \"/opt/autotick-bot/conda_forge_tick/recipe_parser/_parser.py\", line 525, in __init__",
   "    self.meta = self._parser.load(\"\".join(lines))",
   "                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^",
   "  File \"/opt/conda/envs/cf-scripts/lib/python3.11/site-packages/ruamel/yaml/main.py\", line 451, in load",
   "    return constructor.get_single_data()",
   "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^",
   "  File \"/opt/conda/envs/cf-scripts/lib/python3.11/site-packages/ruamel/yaml/constructor.py\", line 114, in get_single_data",
   "    node = self.composer.get_single_node()",
   "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^",
   "  File \"/opt/conda/envs/cf-scripts/lib/python3.11/site-packages/ruamel/yaml/composer.py\", line 72, in get_single_node",
   "    document = self.compose_document()",
   "               ^^^^^^^^^^^^^^^^^^^^^^^",
   "  File \"/opt/conda/envs/cf-scripts/lib/python3.11/site-packages/ruamel/yaml/composer.py\", line 94, in compose_document",
   "    node = self.compose_node(None, None)",
   "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^",
   "  File \"/opt/conda/envs/cf-scripts/lib/python3.11/site-packages/ruamel/yaml/composer.py\", line 130, in compose_node",
   "    node = self.compose_mapping_node(anchor)",
   "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^",
   "  File \"/opt/conda/envs/cf-scripts/lib/python3.11/site-packages/ruamel/yaml/composer.py\", line 211, in compose_mapping_node",
   "    item_value = self.compose_node(node, item_key)",
   "                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^",
   "  File \"/opt/conda/envs/cf-scripts/lib/python3.11/site-packages/ruamel/yaml/composer.py\", line 130, in compose_node",
   "    node = self.compose_mapping_node(anchor)",
   "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^",
   "  File \"/opt/conda/envs/cf-scripts/lib/python3.11/site-packages/ruamel/yaml/composer.py\", line 204, in compose_mapping_node",
   "    while not self.parser.check_event(MappingEndEvent):",
   "              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^",
   "  File \"/opt/conda/envs/cf-scripts/lib/python3.11/site-packages/ruamel/yaml/parser.py\", line 141, in check_event",
   "    self.current_event = self.state()",
   "                         ^^^^^^^^^^^^",
   "  File \"/opt/conda/envs/cf-scripts/lib/python3.11/site-packages/ruamel/yaml/parser.py\", line 614, in parse_block_mapping_key",
   "    raise ParserError(",
   "",
   "'",
   ""
  ]
 },
 "pinning_version": "2023.09.26.15.03.47",
 "pre_pr_migrator_attempts": {},
 "pre_pr_migrator_status": {},
 "smithy_version": "3.26.2"
}